x-pack/plugin/security/qa/jwt-realm/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/jwt/JwtWithUnavailableSecurityIndexRestIT.java

```diff
@@ -0,0 +1,342 @@
+        final URL resource = JwtWithUnavailableSecurityIndexRestIT.class.getResource(name);
+        if (resource == null) {
+            throw new FileNotFoundException("Cannot find classpath resource " + name);
+        }
+        return PathUtils.get(resource.toURI());
+    }
+
+    @Override
+    protected String getTestRestCluster() {
+        return cluster.getHttpAddresses();
+    }
+
+    @Override
+    protected String getProtocol() {
+        return "https";
+    }
+
+    @Override
+    protected Settings restAdminSettings() {
+        String token = basicAuthHeaderValue("admin_user", new SecureString("admin-password".toCharArray()));
+        return Settings.builder().put(ThreadContext.PREFIX + ".Authorization", token).put(restSslSettings()).build();
+    }
+
+    @Override
+    protected Settings restClientSettings() {
+        return Settings.builder().put(super.restClientSettings()).put(restSslSettings()).build();
+    }
+
+    private Settings restSslSettings() {
+        return Settings.builder().put(CERTIFICATE_AUTHORITIES, httpCertificateAuthority).build();
+    }
+
+    protected TestSecurityClient getAdminSecurityClient() {
+        if (adminSecurityClient == null) {
+            adminSecurityClient = new TestSecurityClient(adminClient());
+        }
+        return adminSecurityClient;
+    }
+
+    @AnnotationTestOrdering.Order(10)
+    public void testRoleMappingWithoutCacheFailsWithoutAccessToSecurityIndex() throws Exception {
+        final String dn = randomDn();
+
+        final String rules = Strings.format("""
+            { "all": [
+                { "field": { "realm.name": "jwt1" } },
+                { "field": { "dn": "%s" } }
+            ] }
+            """, dn);
+
+        final List<String> roles = randomRoles();
+        final String roleMappingName = createRoleMapping(roles, rules);
+        final String principal = randomPrincipal();
+
+        try {
+            {
+                final SignedJWT jwt = buildAndSignJwt(principal, dn, Instant.now());
+
+                final Map<String, Object> response = getSecurityClient(jwt).authenticate();
+
+                assertAuthenticationHasUsernameAndRoles(response, principal, roles);
+            }
+
+            makeSecurityIndexUnavailable();
+
+            {
+                final SignedJWT jwt = buildAndSignJwt(principal, dn, Instant.now());
+
+                final Map<String, Object> response = getSecurityClient(jwt).authenticate();
+
+                assertAuthenticationHasUsernameAndRoles(response, principal, List.of());
+            }
+        } finally {
+            makeSecurityIndexAvailable();
+            deleteRoleMapping(roleMappingName);
+
+            // Enable last load caching for next test (setting is not dynamic and requires cluster restart)
+            mutableSettingsForLastLoadCache.put("xpack.security.authz.store.role_mappings.last_load_cache.enabled", "true");
```

Comment:
This feels like a hack to me - there's code in 1 test method that's needed to make another method work.
That means reproductions are going to always fail (you have to run the whole test to get a working reproduction, you can't run the failing method).
To be honest, any test with Ordering gives me bad vibes.

If it were me I'd make it all one test method with a cluster.restart() in it.

Reply:
Fair! I'll tinker with this to make it work in one test method.